src/org/omegat/gui/matches/MatchesVarExpansion.java

```diff
@@ -274,11 +279,29 @@ public String expandVariables(NearString match) {
     }
 
     public Result apply(NearString match, int id) {
+        /**
+         * A sorted map that ensures styled replacements are performed in the
+         * order of appearance.
+         */
+        Map<Integer, Replacer> styledMatch = new TreeMap<>();
         Result result = new Result();
-        styledComponents.clear();
+        boolean isMixedOrientation = BiDiUtils.isMixedOrientationProject();
+        boolean isSourceRtl = false;
+        boolean isTargetRtl = false;
+        if (isMixedOrientation) {
+            isSourceRtl = BiDiUtils.isSourceLangRtl();
+            isTargetRtl = BiDiUtils.isTargetLangRtl();
+        }
 
         // Variables
-        result.text = this.expandVariables(match);
+        result.text = expandVariables(match);
+        if (isMixedOrientation && isTargetRtl) {
+            result.text = result.text.replace(VAR_TARGET_TEXT, BiDiUtils.addRtlBidiAround(match.translation));
+        } else if (isMixedOrientation) {
+            result.text = result.text.replace(VAR_TARGET_TEXT, BiDiUtils.addLtrBidiAround(match.translation));
+        } else {
+            result.text = result.text.replace(VAR_TARGET_TEXT, match.translation);
+        }
```

Comment:
why put expand of VAR_TARGET_TEXT in `expandVariables` method?

Reply:
`expandVariables` is public method and used in 5 places.
It expands `VAR_TARGET_TEXT` in previous code but now It does not expand in the method.
It may lead a bug in another place.